File: backend/src/main/java/com/rih/service/IngredientService.java
package com.rih.service;

import com.rih.entity.Ingredient;
import com.rih.entity.Recipe;

import java.util.List;
import java.util.Set;

public interface IngredientService {

    Ingredient findByName(String name);
    Ingredient saveIngredient(Ingredient ingredient);
    void saveIngredients(Set<Ingredient> ingredients, Long recipeId);

    public List<Recipe> ingredientMatcher(List<Ingredient> ingredients);

}

File: frontend/src/app/services/ingredient.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Ingredient } from '../common/ingredient';

@Injectable({
  providedIn: 'root'
})
export class IngredientService {
  private baseUrl = 'http://ec2-3-27-173-145.ap-southeast-2.compute.amazonaws.com:8081/rih';
  // private baseUrl = 'http://localhost:8080/rih';


  constructor(private httpClient: HttpClient) { }


  getAllIngredients(): Observable<Ingredient[]>{
    return this.httpClient.get<Ingredient[]>(this.baseUrl+'/ingredients');
  }

  saveIngredient(ingredientData: Ingredient):Observable<Ingredient>{
    return this.httpClient.post<Ingredient>(this.baseUrl+'/ingredients/save',ingredientData);

  } 

  // @PostMapping("/save/{recipeId}/ingredient")
  saveIngredients(ingredients: String, recipeId: number): Observable<any>{
    return this.httpClient.post<String>(this.baseUrl+`/ingredients/save/${recipeId}/ingredient`, ingredients);
  }




}
